File: VacationCalendar.Api/Responses/VacationPeriod/GetVacationPeriodsByDatesResponse.cs
namespace VacationCalendar.Api.Responses.VacationPeriod
{
    using VacationCalendar.Api.Dtos;

    public class GetVacationPeriodsByDatesResponse
    {
        public ICollection<VacationPeriodDto> VacationPeriods { get; set; }
    }
}

File: VacationCalendarAngular/src/app/dtos/responses/get-vacation-periods-by-dates-response.ts
import { VacationPeriodDto } from "../vacation-period-dto";

export class GetVacationPeriodsByDatesResponse {
    public vacationPeriods: Array<VacationPeriodDto>
    constructor(
        vacationPeriods: Array<VacationPeriodDto>
    ){
        // map models from response to dtos - dates not mapped properly
        this.vacationPeriods = new Array<VacationPeriodDto>();

        vacationPeriods.forEach(vacationPeriod =>{
            this.vacationPeriods.push(new VacationPeriodDto(
                vacationPeriod.id,
                vacationPeriod.userId,
                vacationPeriod.firstName,
                vacationPeriod.lastName,
                vacationPeriod.notes,
                vacationPeriod.from.toString(),
                vacationPeriod.to.toString()))
            }
        );
    } 
}